runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -61,11 +63,14 @@ void incrementLapCount(Client client)
 		} while (remainingXp < 0);
 
 		lapsTillLevel = remainingXp > 0 ? (int) Math.ceil(remainingXp / course.getTotalXp()) : 0;
```

Comment:
This should just be merged together with goal XP and show next goal based on player's goal settings instead of displaying both